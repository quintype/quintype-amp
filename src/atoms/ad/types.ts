import { ReactElement } from "react";

export interface AdTypes {
  type: string;
  width: string | number;
  height: string | number;
  children?: JSX.Element[] | JSX.Element | ReactElement | HTMLElement;
  [key: string]: any;
}
