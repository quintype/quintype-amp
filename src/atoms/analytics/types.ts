export interface AnalyticsProps {
  type?: string;
  children?: JSX.Element | JSX.Element[];
  "data-credentials"?: string;
  config?: string;
  targets?: object;
  requests?: object;
  vars?: object;
  extraUrlParams?: object;
  triggers?: object;
  transport?: object;
  id?: string;
}
