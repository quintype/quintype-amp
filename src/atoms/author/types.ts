import { Author } from "../../types/story";

export interface AuthorProps {
  authors: Author[];
  prepend?: string | React.Component | React.ReactElement;
  inlineStyles?: object;
}
