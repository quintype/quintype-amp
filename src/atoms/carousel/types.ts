import { ReactElement } from "react";
import { Common } from "../common-types";
import { DefaultTheme, CSSObject } from "styled-components";

export interface CarouselTypes extends Common {
  type: "slides" | "carousel";
  id?: string;
  controls?: boolean;
  "data-next-button-aria-label"?: string;
  "data-prev-button-aria-label"?: string;
  "data-button-count-format"?: string;
  autoplay?: boolean | string;
  delay?: string;
  loop?: boolean;
  slide?: string;
  children?: JSX.Element[] | JSX.Element | ReactElement;
  style?: CustomStyles;
  theme?: DefaultTheme;
}

interface CustomStyles {
  carousel?: (theme) => CSSObject;
}
