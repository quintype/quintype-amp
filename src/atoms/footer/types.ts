import { CSSObject } from "styled-components";

export interface FooterTypes {
  text?: string;
  children?: JSX.Element[] | JSX.Element | React.ReactChildren | React.ReactChild;
  style?: (props) => CSSObject;
  showPoweredByQt?: boolean | (() => boolean);
}
