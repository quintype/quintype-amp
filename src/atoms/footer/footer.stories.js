import * as React from "react";
import { storiesOf } from "@storybook/react";
import { Footer } from "./footer";
import { storyWithManyJsEmbeds, config } from "../../__fixtures__";
import Layout from "../layout";

storiesOf("Footer", module)
  .addDecorator((story) => (
    <Layout config={config} story={storyWithManyJsEmbeds}>
      {story()}
    </Layout>
  ))
  .add("No Footer Text", () => <Footer />)
  .add("With Footer Text", () => (
    <Footer text="Copyright Â© 2020 Lorem Ipsum Media Private Limited. All Rights Reserved" />
  ))
  .add("With custom components", () => (
    <Footer>
      <p>This is a custom footer</p>
      <p>Powered by Quintype AMP</p>
    </Footer>
  ));
