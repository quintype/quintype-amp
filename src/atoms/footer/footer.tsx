import React, { Fragment } from "react";
import styled, { css } from "styled-components";
import get from "lodash.get";
import { FooterTypes } from "./types";
import { genStyles } from "../../helpers/gen-styles";
import { withTheme } from "styled-components";
import { withConfig } from "../../context";

const baseStyles = css`
  display: flex;
  padding: 8px 4px;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  min-height: 50px;
  font-family: ${(props) => props.theme.font.family.secondary};
  background-color: ${(props) => props.theme.color.footerBackground};
  color: ${(props) => props.theme.color.footerTextColor};
`;

export const PoweredBy = styled.a`
  text-decoration: none;
  color: ${(props) => props.theme.color.footerTextColor};
  font-size: ${(props) => props.theme.font.size.xxs};
`;

const StyledFooter = styled("footer")<FooterTypes>`
  ${(props) => genStyles(baseStyles, props.style, props)}
`;

export const DefaultFooter = ({ text, children, style, config }: FooterTypes) => {
  let showPoweredByQt: boolean | ((config) => boolean) = get(
    config,
    ["opts", "featureConfig", "showPoweredByQt"],
    true
  );

  showPoweredByQt = typeof showPoweredByQt === "function" ? showPoweredByQt(config) : showPoweredByQt;
  return (
    <StyledFooter style={style}>
      {children ? (
        children
      ) : (
        <Fragment>
          {text && <p>{text}</p>}
          {showPoweredByQt && (
            <PoweredBy href="https://www.quintype.com/" rel="noreferrer noopener" target="_blank">
              Powered by Quintype
            </PoweredBy>
          )}
        </Fragment>
      )}
    </StyledFooter>
  );
};

const BaseFooter = ({ text, children, style, config  }: FooterTypes) => {
  const footerRender = get(config, ["opts", "render", "footerRender"], null);

  return footerRender ? (
    footerRender({ text, children, style, config })
  ) : (
    <DefaultFooter text={text} children={children} style={style} config={config} />
  );
};

export { BaseFooter };

const Footer = withConfig(withTheme(BaseFooter));
export { Footer };
