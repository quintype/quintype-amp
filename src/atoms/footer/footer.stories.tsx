import * as React from "react";
import { storiesOf } from "@storybook/react";
import { Footer, BaseFooter } from "./footer";
import { config, textStory } from "../../__fixtures__";
import { Layout } from "../layout";

storiesOf("Footer", module)
  .addDecorator((story) => (
    <Layout story={textStory} config={config}>
      {story()}
    </Layout>
  ))
  .add("No Footer Text", () => <Footer />)
  .add("With Footer Text", () => (
    <Footer text="Copyright Â© 2020 Lorem Ipsum Media Private Limited. All Rights Reserved" />
  ))
  .add("With custom components", () => (
    <Footer>
      <p>This is a custom footer</p>
      <p>Powered by Quintype AMP</p>
    </Footer>
  ))
  .add("With custom style", () => {
    const customStyles: any = {};
    customStyles.footer = (theme) => ({
      backgroundColor: `${theme.color.primaryColor}`
    });
    return <Footer text="I'm a footer" style={customStyles} />;
  })
  .add("With custom style and custom theme", () => {
    const customStyles: any = {};
    customStyles.footer = (theme) => ({
      backgroundColor: `${theme.color.primaryColor}`
    });
    return (
      <BaseFooter
        text="I'm a footer"
        style={customStyles}
        theme={{
          color: {
            primaryColor: "pink"
          }
        }}
      />
    );
  });
