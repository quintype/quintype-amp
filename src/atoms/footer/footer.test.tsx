import React from "react";
import { BaseFooter, PoweredBy, DefaultFooter } from "./footer";
import { shallow } from "enzyme";
import { Config } from "../../types/config";
import { config } from "../../../src/__fixtures__";

describe("Footer", () => {
  it("should render", () => {
    const wrapper = shallow(<DefaultFooter />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should render children", () => {
    const wrapper = shallow(
      <DefaultFooter>
        <amp-img
          src="https://images.unsplash.com/photo-1558980395-be8a5fcb4251?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60"
          width="400"
          height="300"
          alt="a sample image"
        />
      </DefaultFooter>
    );
    expect(wrapper.find("amp-img").length).toBe(1);
  });
  it("should render powered by quintype when showPoweredByQt attribute is not passed", () => {
    const wrapper = shallow(<DefaultFooter config={config} />);
    expect(wrapper.find(PoweredBy).exists()).toBeTruthy();
  });
  it("should render powered by quintype when showPoweredByQt attribute is passed as a function with truthy value", () => {
    const defaultConfig: Config = config;
    defaultConfig.opts = {
      featureConfig: {
        showPoweredByQt: () => true
      }
    };

    const wrapper = shallow(<DefaultFooter config={defaultConfig} />);
    expect(wrapper.find(PoweredBy).exists()).toBeTruthy();
  });
  it("should not render powered by quintype when showPoweredByQt attribute is passed as a function with falsy value", () => {
    const defaultConfig: Config = config;
    defaultConfig.opts = {
      featureConfig: {
        showPoweredByQt: () => false
      }
    };

    const wrapper = shallow(<DefaultFooter config={defaultConfig} />);
    expect(wrapper.find(PoweredBy).exists()).toBeFalsy();
  });
  it("should render powered by quintype when showPoweredByQt attribute is passed as a boolean with truthy value", () => {
    const defaultConfig: Config = config;
    defaultConfig.opts = {
      featureConfig: {
        showPoweredByQt: true
      }
    };

    const wrapper = shallow(<DefaultFooter config={defaultConfig} />);
    expect(wrapper.find(PoweredBy).exists()).toBeTruthy();
  });
  it("should not render powered by quintype when showPoweredByQt attribute is passed as a boolean with falsy value", () => {
    const defaultConfig: Config = config;
    defaultConfig.opts = {
      featureConfig: {
        showPoweredByQt: false
      }
    };

    const wrapper = shallow(<DefaultFooter config={defaultConfig} />);
    expect(wrapper.find(PoweredBy).exists()).toBeFalsy();
  });
  it("should call footerRender prop when passed to opts", () => {
    const footerRender = jest.fn();
    const modifiedConfig = {
      ...config,
      opts: { render: { footerRender } }
    };
    const wrapper = shallow(<BaseFooter config={modifiedConfig} />);
    expect(footerRender.mock.calls.length).toBe(1);
    expect(wrapper.find(DefaultFooter).length).toBe(0);
  });
});
