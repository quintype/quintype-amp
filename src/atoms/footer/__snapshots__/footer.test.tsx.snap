// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render 1`] = `
<styled.footer>
  <styled.a
    href="https://www.quintype.com/"
    rel="noreferrer noopener"
    target="_blank"
  >
    Powered by Quintype
  </styled.a>
</styled.footer>
`;
